test/dialect/postgresql/test_reflection.py

```diff
@@ -1524,31 +1525,55 @@ def test_reflect_unique_index(self):
     def test_reflect_check_constraint(self):
         meta = self.metadata
 
-        cc_table = Table(
-            "pgsql_cc",
-            meta,
-            Column("a", Integer()),
-            CheckConstraint("a > 1 AND a < 5", name="cc1"),
-            CheckConstraint("a = 1 OR (a > 2 AND a < 5)", name="cc2"),
-        )
+        sql = text("""\
+            CREATE OR REPLACE FUNCTION is_positive(
+                x integer DEFAULT '-1'::integer)
+                RETURNS boolean
+                LANGUAGE 'plpgsql'
+                COST 100
+                VOLATILE 
+            AS $BODY$BEGIN
+                RETURN x > 0;
+            END;$BODY$;
+        """)
+        testing.db.execute(sql)
```

Comment:
associate the DDL object with the metadata that the decorator gives us:

```
event.listen(
    self.metadata,
    "before_create",
    DDL("CREATE OR REPLACE FUNCTION ...")
)
```

at the same time, associate the "drop" DDL also:

`event.listen(self.metadata, "after_drop", DDL("DROP FUNCTION is_positive"))`